libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ProbabilityDistributionSplineChart.tsx

```diff
@@ -1,23 +1,30 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { getTheme, IChoiceGroupOption } from "@fluentui/react";
+import { getTheme, IChoiceGroupOption, Spinner } from "@fluentui/react";
 import {
   BasicHighChart,
   calculateSplinePlotDataFromErrorCohort,
   defaultModelAssessmentContext,
   ErrorCohort,
-  ModelAssessmentContext
+  ModelAssessmentContext,
+  ifEnableLargeData,
+  IProbabilityBinCount,
+  Cohort
 } from "@responsible-ai/core-ui";
 import { localization } from "@responsible-ai/localization";
+import _ from "lodash";
 import React from "react";
 
 interface IProbabilityDistributionSplineChartProps {
   selectedCohorts: ErrorCohort[];
   probabilityOption?: IChoiceGroupOption;
 }
 
-class IProbabilityDistributionSplineChartState {}
+interface IProbabilityDistributionSplineChartState {
+  probabilityBinCountArray?: Array<IProbabilityBinCount[] | undefined>;
+  selectedCohorts: ErrorCohort[];
```

Comment:
Why do we need this as state? I see updated value is send in props and both times we set directly from props in state. Can we still have this just as prop and use in render?

Reply:
Yes, we need it in the state since the number of entries in probabilityBinCountArray need to equal to selectedCohorts. If we use selectedCohorts from props then there will be mismatch in the number of entries and I see a crash.